rust/noosphere-ns/src/bin/bootstrap_nns/cli.rs

```diff
@@ -0,0 +1,50 @@
+#![cfg(not(target_arch = "wasm32"))]
+
+use clap::{Parser, Subcommand};
+use serde::Deserialize;
+use std::path::PathBuf;
+
+#[derive(Parser)]
+#[clap(name = "bootstrap_nns")]
+pub struct CLI {
+    #[clap(subcommand)]
+    pub command: CLICommand,
+}
+
+#[derive(Subcommand)]
+pub enum CLICommand {
+    Install,
+    Uninstall,
+    /// Run DHT nodes in the Noosphere Name System.
+    Run {
+        /// The path to the bootstrap configuration, a TOML file, containing
+        /// entries for keyname/port pairs.
+        #[clap(short, long)]
+        config: Option<PathBuf>,
+
+        /// If no configuration path provided, the name of the Noosphere keypair to use
+        /// stored in `~/.noosphere/keys/`.
+        #[clap(short, long)]
+        key: Option<String>,
+
+        /// If no configuration path provided, the listening port of this DHT node.
+        #[clap(short, long)]
+        port: Option<u16>,
+
+        /// If no configuration path provided, a flag to create key with name `--key`
+        /// if one does not already exist.
+        #[clap(long)]
+        create_key_if_needed: bool,
```

Comment:
`create_key_if_needed` is a temporary solution to provide some key gen capabilities here without pulling in `orb`. Open to suggestions here, I'm mostly indifferent, wanting to keep it simple (as opposed to a whole `key` subcommand)

Reply:
In the CLI and in our Rust APIs (including FFI), we put the burden on the user to create a key. This makes more sense in those cases: for users, there is usually a user-given ID (e.g., a username) given that the key should be associated with so that when they perform a future TouchID authorization (or equivalent action), they see the name of the key somewhere in the UI.

I'm not as opinionated for the bootstrap nodes, but if you want to be consistent with the CLI then the key should just be assumed to be created in advance (or else it's an error).